File: cci16.01-swap-numbers-lcci/python/Solution.py
class Solution:
    def swapNumbers(self, numbers: List[int]) -> List[int]:
        if numbers[0] == numbers[1]:
            return numbers
        numbers[0] ^= numbers[1]
        numbers[1] ^= numbers[0]
        numbers[0] ^= numbers[1]
        return numbers

File: cci16.01-swap-numbers-lcci/cpp/Solution.cpp
class Solution {
public:
    vector<int> swapNumbers(vector<int>& numbers) {
        if (numbers[0] == numbers[1]) return numbers;
        numbers[0] ^= numbers[1];
        numbers[1] ^= numbers[0];
        numbers[0] ^= numbers[1];
        return numbers;
    }
};
